lib/src/test/java/com/auth0/jwt/impl/PayloadDeserializerTest.java

```diff
@@ -220,6 +220,19 @@ public void shouldGetDateWhenParsingNumericNode() throws Exception {
     }
 
     @Test
+    public void shouldGetLargeDateWhenParsingNumericNode() throws Exception {
+        Map<String, JsonNode> tree = new HashMap<>();
+        long seconds = Integer.MAX_VALUE + 10000L;
+        LongNode node = new LongNode(seconds);
+        tree.put("key", node);
+
+        Date date = deserializer.getDateFromSeconds(tree, "key");
+        assertThat(date, is(notNullValue()));
+        assertThat(date.getTime(), is(seconds * 1000));
+        assertThat(date.getTime(), is(2147493647L * 1000));
```

Comment:
`Integer` is always 32 bits on java?
it's weird how java makes you explicitly cast to `(Object)` sometimes, but when you sum a long to an int it doesn't care.